packages/vite-node/src/client.ts

```diff
@@ -215,6 +215,18 @@ export class ViteNodeRunner {
     return [resolvedId, fsPath]
   }
 
+  async resolveUrl(id: string, importee?: string) {
+    const resolveKey = `resolve:${id}`
+    // put info about new import as soon as possible, so we can start tracking it
+    this.moduleCache.set(resolveKey, { resolving: true })
```

Comment:
ModuleCache's `set` will apply a normalization. Ideally, in this case, we could expose `_set` `_delete` without normalization to be used here, and avoid extra normalization costs.